R/run.R

```diff
@@ -241,6 +256,56 @@ global_setup <- function(lib_path = NULL, cpu_count = NULL, mem_alloc = NULL, te
   script
 }
 
+#' Assemble metadata for a benchmark run
+#'
+#' @param name Benchmark name, i.e. `bm$name`
+#' @param params Named list of parameters for the individual run, i.e. the case
+#' @param cpu_count Number of CPUs allocated
+#' @param n_iter Number of iterations
+#'
+#' @keywords internal
+assemble_metadata <- function(name, params, cpu_count, n_iter) {
+  tags <- params
+  tags[["dataset"]] <- params$source
+  tags[["source"]] <- NULL
+  tags[["cpu_count"]] <- cpu_count
+  tags[["language"]] <- "R"
+
+  arrow_info <- arrow::arrow_info()
+  info <- list(
+    arrow_version = arrow_info$build_info$cpp_version,
+    arrow_compiler_id = arrow_info$build_info$cpp_compiler,
+    arrow_compiler_version = arrow_info$build_info$cpp_compiler_version,
+    benchmark_language_version = version[['version.string']],
+    arrow_version_r = as.character(arrow_info$version)
+  )
+
+  context <- list(
+    arrow_compiler_flags = arrow_info$build_info$cpp_compiler_flags,
```

Comment:
Until I saw this here, I hadn't totally put together how much this could actually differ from what we see in conbench (depending on how exactly we install the arrow R package, if we install pointing to libarrow that was build during orchestration it'll be the same, but if we install with a static lib, these flags could actually be quite different than the ones we see from the conbench|benchmarks utilities we are using now). Very nice!